contrib/flo-bigquery/src/main/java/com/spotify/flo/contrib/bigquery/BigQueryLookup.java

```diff
@@ -55,4 +60,40 @@ Spec lookup(TableId tableId) {
       return bq.tableExists(tableId) ? Optional.of(tableId) : Optional.empty();
     }
   }
+
+  static class LookupLatestDailySpec implements Spec {
+
+    private static final long serialVersionUID = 1L;
+
+    final String project;
+    final String dataset;
+    final String tableName;
+    final Date start;
+    final int lookBackDays;
+
+    LookupLatestDailySpec(String project, String dataset, String tableName, Date start, final int lookBackDays) {
+      this.project = project;
```

Comment:
null checks

Reply:
should they be added to only the public-facing API in BigQueryTasks or to both that and the constructor here as well?

Reply:
went ahead with the former but lmk if that's the wrong approach and will swap!